### PR TITLE
feature/codeception-upgrade-to-v5

### PR DESCRIPTION
**Pull request by ggiak**
Following up to https://github.com/Codeception/Codeception/blob/5.0/src/Codeception/Module.php

This is the complete Pull request to support Codeception v5
as well as some restructuring quite few things for the project to make it compatible to php8 
at the same time there are some structural changes on the project to better match with the rest of the codeception modules.

Introduced folder structure as follows:

src/Codeception

- src/Codeception/Exception
- src/Codeception/Module
- src/Codeception/Report

Additionally this also includes:

1. compatibility with php8.2
2. compatibility with codeception 5.0
3. phpunit 9.5
4. Strict types
5. composer json updates
6. new module version (v5) to match with the parent one
7. new docker image for php8
8. Namespaces
9. and more...

### Testing

`docker-compose up --abort-on-container-exit --build`

Results:

image

### CODE REVIEW

**Comment by ggiak** [comment]
@buresmi7, @Arhell

**Comment by koehnlein** [comment]
I fixed the automated test in another pull request: https://github.com/ggiak/VisualCeption/pull/3

I don't know how and if I can update this pull request directly. But maybe someone else could have a look.

File: src/Codeception/Module/VisualCeption.php [hunk 1]
```diff
@@ -22,58 +26,90 @@
  */
 class VisualCeption extends CodeceptionModule implements MultiSession
 {
-    protected $config = [
+    /**
+     * @var array
+     */
```
Comment (buresmi7) [inline src/Codeception/Module/VisualCeption.php:31]: I think it duplucates the type definition, it can be removed?

**Comment by buresmi7** [comment]
> I fixed the automated test in another pull request: [ggiak#3](https://github.com/ggiak/VisualCeption/pull/3)
> 
> I don't know how and if I can update this pull request directly. But maybe someone else could have a look.

@ggiak should approve your PR

BTW, good job guys ;)

**Comment by buresmi7** [comment]
Should I squash this PR? this action probably remove @koehnlein as the author of part of the code, on the other hand there is 18 commits... what do you think @ggiak @koehnlein ?

**Comment by ggiak** [comment]
@buresmi7 I am ok with it, though if @koehnlein wishes we can rebase interactively.

**Comment by koehnlein** [comment]
@buresmi7 @ggiak Beeing mentioned as contributor would be nice. But if it causes trouble, it's not necessary.

**Comment by ggiak** [comment]
@buresmi7, @koehnlein, I rebased, left 1 commit with co-authored-by & commit hash. 

This should do the trick :)

**Comment by koehnlein** [comment]
@buresmi7 @ggiak I'm not sure, if this works. But im fine with merging. I do not want to block this great improvement only because of the question, if I would be listed somewhere or not. 😄

**Pull request merged**